source/beamline/AMPVControl.cpp

```diff
@@ -213,7 +213,7 @@ AMControl::FailureExplanation AMPVControl::move(double setpoint) {
 void AMPVControl::onNewFeedbackValue(double) {
 
 	// If we're not in the middle of a move, don't really care about changing values.
-	if(!moveInProgress_)
+	if(!moveInProgress_ && !completionTimer_.isActive())
```

Comment:
@dretrex for the crystal selection, it needs a long to move. The value of feedback will change twice during the process. When the final value is changed, the `moveInProgress_` is False already. But we still have the `completionTimer_` active (60s timout for this case). 

With this check, we will be able to have the complete signal ASAP. This is our original design as well (in the codes below, we stopped the timer. But it will only works when the move is quick enough).